modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_9_0/test/DDMDataProviderInstanceUpgradeProcessTest.java

```diff
@@ -0,0 +1,143 @@
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class DDMDataProviderInstanceUpgradeProcessTest
+	extends BaseCTUpgradeProcessTestCase {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+		DDMForm ddmForm = DDMFormTestUtil.createDDMForm(
+			RandomTestUtil.randomString());
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getGroupId());
+
+		_ddmFormValues = DDMFormValuesTestUtil.createDDMFormValues(ddmForm);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		_resourceLocalService.deleteResource(
+			_group.getCompanyId(), DDMDataProviderInstance.class.getName(),
+			ResourceConstants.SCOPE_INDIVIDUAL, _dataProviderInstanceId);
+	}
+
+	@Override
+	protected CTModel<?> addCTModel() throws Exception {
+		DDMDataProviderInstance dataProviderInstance =
+			_ddmDataProviderInstanceLocalService.addDataProviderInstance(
+				TestPropsValues.getUserId(), _group.getGroupId(),
+				RandomTestUtil.randomLocaleStringMap(), null, _ddmFormValues,
+				"rest", _serviceContext);
+
+		_dataProviderInstanceId =
+			dataProviderInstance.getDataProviderInstanceId();
+
+		return dataProviderInstance;
+	}
+
+	@Override
+	protected CTService<?> getCTService() {
+		return _ddmDataProviderInstanceLocalService;
+	}
+
+	@Override
+	protected void runUpgrade() throws Exception {
+		UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+			_upgradeStepRegistrator, _CLASS_NAME);
+
+		upgradeProcess.upgrade();
+	}
+
+	@Override
+	protected CTModel<?> updateCTModel(CTModel<?> ctModel) throws Exception {
+		DDMDataProviderInstance ddmDataProviderInstance =
+			(DDMDataProviderInstance)ctModel;
+
+		DDMForm ddmForm = DDMFormTestUtil.createDDMForm(
+			RandomTestUtil.randomString());
+
+		_ddmFormValues = DDMFormValuesTestUtil.createDDMFormValues(ddmForm);
```

Comment:
@p-albuquerque why do we need to create a new `DDMForm` here? Can't we just create new `DDMFormValues` with the same `DDMForm`?